### PR TITLE
boards: added ThePeach K1/R1 Flight Controller

### PR DESCRIPTION
**Pull request by ThePeach-kr**
Dear PX4 Team,

We developed a Flight Controller.
I would like to add our K1/R1 Flight Controller code to the PX4-Autopilot.
Also, rebase on Nuttx 10.3+

Related PR
https://github.com/PX4/PX4-Autopilot/pull/20176

Thanks.

### CODE REVIEW

**Comment by junwoo091400** [comment]
Is this available for purchase online by any chance? Would you be interested in adding this board to the PX4 Documentation?

File: boards/thepeach/k1/nuttx-config/include/board.h [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+#endif
+
+#include <stm32.h>
+
+/************************************************************************************
+ * Definitions
+ ************************************************************************************/
+
+/* Clocking *************************************************************************/
+/* The PX4FMUV2 uses a 24MHz crystal connected to the HSE.
```
Comment (davids5) [inline boards/thepeach/k1/nuttx-config/include/board.h:58]: Name should be this HW

File: boards/thepeach/k1/nuttx-config/include/board.h [hunk 2]
```diff
@@ -0,0 +1,271 @@
+/************************************************************************************
+ * board.h
+ * include/arch/board/board.h
+ *
+ *   Copyright (C) 2009, 2016 Gregory Nutt. All rights reserved.
```
Comment (davids5) [inline boards/thepeach/k1/nuttx-config/include/board.h:5]: Add 2022

File: boards/thepeach/r1/nuttx-config/include/board.h [hunk 3]
```diff
@@ -0,0 +1,266 @@
+/************************************************************************************
+ * board.h
+ * include/arch/board/board.h
+ *
+ *   Copyright (C) 2009, 2016 Gregory Nutt. All rights reserved.
```
Comment (davids5) [inline boards/thepeach/r1/nuttx-config/include/board.h:5]: Add 2022

File: boards/thepeach/r1/nuttx-config/include/board.h [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+#endif
+
+#include <stm32.h>
+
+/************************************************************************************
+ * Definitions
+ ************************************************************************************/
+
+/* Clocking *************************************************************************/
+/* The PX4FMUV2 uses a 24MHz crystal connected to the HSE.
```
Comment (davids5) [inline boards/thepeach/r1/nuttx-config/include/board.h:58]: Name as this HW

**Review by davids5 — CHANGES_REQUESTED** [review]
See change requests. 

As an aside question: How are R1 and K1 different? Gould it be one binary and uses runtime detection?

**Comment by ThePeach-kr** [comment]
> See change requests.
> 
> As an aside question: How are R1 and K1 different? Gould it be one binary and uses runtime detection?

R1 is integrated with Raspberry Pi CM3+, unlike K1. Other than this, there's no big difference.
but, There is no version detection pin, so it will be difficult to make it run as one binary.

Thank you.

**Review by davids5 — APPROVED** [review]

**Comment by ThePeach-kr** [comment]
@hamishwillee 
I would appreciate it if you could add our boards to the sidebar!
I appreciate your work on this :smile:

**Pull request merged**